include/alpaka/queue/QueueCpuNonBlocking.hpp

```diff
@@ -73,6 +74,20 @@ namespace alpaka
                     //-----------------------------------------------------------------------------
                     ~QueueCpuNonBlockingImpl() = default;
 
+                    //-----------------------------------------------------------------------------
+                    void enqueue(std::shared_ptr<cpu::ICpuQueue> & iQueue, event::EventCpu & ev) final
+                    {
+                        auto spQueueImpl = std::static_pointer_cast<QueueCpuNonBlockingImpl>(iQueue);
```

Comment:
Do we really need the separate `iQueue` parameter? We should have a `enqueue` and `wait` overload for the non-shared_ptr types. Then we only would have to pass `*this`.

Reply:
I will try it again. I started with that before but somehow I was unhappy with that. I do not know anymore why so I will try it again.